### ISSUE TITLE
Manually provided __init__ in a dataclass is ignored

### ISSUE THREAD

**Issue by benjaminsaljooghi** [microsoft/pyright#203]
**Describe the bug**
Dataclasses allow you to specify `init=False` or manually provide an `__init__` to customize initialization of a dataclass. When provided manually, pyright seems to ignore it and instead assumes the dataclass is initialized based on its fields.

**To Reproduce**
```Python
@dataclass(init=False) # This argument is optional given that there is a manually provided __init__ below
class A:
    x: int
    x_squared: int

    def __init__(self, x: int):
        self.x = x
        self.x_squared = x ** 2

a = A(3)
```

**Expected behavior**
Pylint does not flag any issues.

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/15173618/63067228-45740b80-bf51-11e9-9818-508608bd4471.png)

**VS Code extension or command-line**
VS Code extension version 1.0.55

**Additional context**
Thank you for this awesome tool! Picks up many things mypy doesn't.

**Comment by erictraut** [comment]
Thanks for the suggestion. This will be fixed in the next published version of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.0.56, which I just published.

**Comment by benjaminsaljooghi** [comment]
Thank you! Works perfectly :)

**Comment by benjaminsaljooghi** [comment]
I have a situation I am using inheritance with dataclasses and the sub-class has a manually provided `__init__` which is ignored by Pyright.

```Python
@dataclass
class A:
    x: int
    
@dataclass
class B(A):
    y: int

    def __init__(self, a: A, y: int):
        self.__dict__ = a.__dict__
        self.y = y

a = A(3)
b = B(a, 5) # Pyright flags an error here
```

It's a fairly exceptional case but I wonder if Pyright could give priority to the manually provided `__init__` rather than the fields of the superclass `A` here?
